ngraph/core/src/op/squeeze.cpp

```diff
@@ -149,28 +149,40 @@ shared_ptr<Node> op::Squeeze::clone_with_new_inputs(const OutputVector& new_args
 namespace squeeze
 {
     template <element::Type_t ET>
-    bool evaluate(const HostTensorPtr& arg0, const HostTensorPtr& out)
+    bool evaluate(const HostTensorPtr& arg0, const HostTensorPtr& arg1, const HostTensorPtr& out)
     {
-        runtime::reference::copy(
-            arg0->get_data_ptr<ET>(), out->get_data_ptr<ET>(), shape_size(out->get_shape()));
+        const auto data_rank = arg0->get_partial_shape().rank().get_length();
+        const auto axes_num = shape_size(arg1->get_shape());
+        auto axes = normalize_axes(
+            "",
+            std::vector<int64_t>(arg1->get_data_ptr<ET>(), arg1->get_data_ptr<ET>() + axes_num),
+            data_rank);
+        std::sort(axes.begin(), axes.end(), [](int64_t a, int64_t b) { return a > b; });
+
+        auto out_shape = arg0->get_shape();
+        for (const auto& axis : axes)
+        {
+            out_shape.erase(out_shape.begin() + axis);
+        }
+        out->set_shape(out_shape);
+
+        runtime::reference::copy(arg0->get_data_ptr<char>(),
+                                 out->get_data_ptr<char>(),
+                                 shape_size(out_shape) * out->get_element_type().size());
         return true;
     }
 
     bool evaluate_squeeze(const HostTensorPtr& arg0,
                           const HostTensorPtr& arg1,
                           const HostTensorPtr& out)
     {
-        auto element_type = arg0->get_element_type();
+        auto element_type = arg1->get_element_type();
 
         bool rc = true;
         switch (element_type)
         {
-            NGRAPH_TYPE_CASE(evaluate_squeeze, i32, arg0, out);
-            NGRAPH_TYPE_CASE(evaluate_squeeze, i64, arg0, out);
-            NGRAPH_TYPE_CASE(evaluate_squeeze, u32, arg0, out);
-            NGRAPH_TYPE_CASE(evaluate_squeeze, u64, arg0, out);
-            NGRAPH_TYPE_CASE(evaluate_squeeze, f16, arg0, out);
-            NGRAPH_TYPE_CASE(evaluate_squeeze, f32, arg0, out);
```

Comment:
Is it okay that you removed these?

Reply:
Yes, the previous version had a switch over the data input type and didn't consider axes at all. In my version we don't mind the data input type (because squeeze is actually doing a copy and in this case we only need to know the input element size), but as I'm considering the axes I need to switch over the possible axes type which are i32 and i64 only.

Reply:
Thanks for explanation!